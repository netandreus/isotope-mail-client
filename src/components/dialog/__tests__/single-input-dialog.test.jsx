import React from 'react';
import {shallow} from 'enzyme/build/index';
import SingleInputDialog from '../single-input-dialog';

const DEFAULT_MOCK_PROPS = {
  visible: true,
  disabled: false,
  titleLabel: 'Dr. POTUS',
  messageLabel: 'Christmas Message',
  inputLabel: 'This is the input',
  inputValue: 'Initial value',
  cancelLabel: 'Cancel',
  cancelAction: jest.fn(),
  okLabel: 'OK',
  okAction: jest.fn()}

describe('SingleInputDialog component test suite', () => {
  test('Snapshot render, should render SingleInputDialog', () => {
    // Given

    // When
    const singleInputDialog = shallow(<SingleInputDialog {...DEFAULT_MOCK_PROPS} />);

    // Then
    expect(singleInputDialog).toMatchSnapshot();
  });
});
