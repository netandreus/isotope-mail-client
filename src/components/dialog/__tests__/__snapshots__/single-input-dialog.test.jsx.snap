// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleInputDialog component test suite Snapshot render, should render SingleInputDialog 1`] = `
<Dialog
  actions={
    Array [
      Object {
        "action": [MockFunction],
        "disabled": false,
        "label": "Cancel",
      },
      Object {
        "action": [Function],
        "disabled": false,
        "label": "OK",
      },
    ]
  }
  className="singleInputDialog"
  containerClassName="container"
  contentClassName="content"
  scrimClick={[Function]}
  title="Dr. POTUS"
  visible={true}
>
  <span>
    Christmas Message
  </span>
  <form
    onSubmit={[Function]}
  >
    <TextField
      autoComplete="off"
      disabled={false}
      fieldClass="mdc-text-field--fullwidth"
      focused={true}
      id="folderName"
      inputClass=""
      label="This is the input"
      labelClass=""
      lineRippleClass=""
      min=""
      minLength={null}
      onChange={[Function]}
      onKeyDown={[Function]}
      onKeyPress={null}
      required={true}
      type="text"
      value="Initial value"
    />
  </form>
</Dialog>
`;
